BDArmory/VesselSpawning/CircularSpawning.cs

```diff
@@ -174,14 +174,54 @@ private IEnumerator SpawnAllVesselsOnceCoroutine(CircularSpawnConfig spawnConfig
             if (BDArmorySettings.VESSEL_SPAWN_RANDOM_ORDER) spawnConfig.craftFiles.Shuffle(); // Randomise the spawn order.
             int spawnedVesselCount = 0; // Reset our spawned vessel count.
             var spawnAirborne = spawnConfig.altitude > 10;
-            bool PinataMode = false;
-            foreach (var craftUrl in spawnConfig.craftFiles)
+            int pinataCount = 0;
+
+            if (BDArmorySettings.TOURNAMENT_NPCS_PER_HEAT > 0)
```

Comment:
This still seems to have a lot of NPC stuff mixed in, which is pretty much all handled in the tournament code now, so it shouldn't need anything here.

I think it'd be better to have the piñata stuff completely separate from the NPC stuff.

The NPCs are currently spawned the same as if they were a regular player (i.e., in the same spawning ring). If we want the piñatas to spawn separately from the other craft, then I think it'd be better to have a separate coroutine that does the spawning and team assignment of the piñatas and/or the other craft. It'd just have to set up a second circular spawn with "kill everything first" set to `false` and a different radius to the other craft and be called as part of the tournament heat execution.